### DISCUSSION TITLE
Rename "admin" to "twill"

### DISCUSSION THREAD

**Discussion by antonioribeiro** [area17/twill#1012] [Ideas]
The name "admin" for Twill is too generic. Being Twill a CMS and not an all purposes administration panel nor a framework for creating management systems, the name "admin" may get in the way of something important for some apps. What I propose here is to change everything publicly named "admin" to **twill**.

Pros:
- Prevent conflicts management classes someone may already have in their apps before installing Twill
- Prevent conflicts with third party admin panels
- Remove ambiguity: easily identify/find code related to Twill. Even if someone is able to rename their code to open some space for Twill, names can still conflit and a simple search for "admin" may be a problem for them
- You can afford this kind of change in a major version (2.0)
- Tests, we are now writing, will help make this change
- Improve Twill's brand awareness

Cons:
- Upgrade may not be as straightforward as we would like
- Upgrade notes would have to be very good

Examples of files and namespaces prone to name clashing today, or which could be improved while renamed to **T/twill**:

```
./routes/admin.php
./public/assets/admin
./app/Http/Requests/Admin
./app/Http/Controllers/Admin

namespace App\Http\Controllers\Admin
namespace App\Http\Requests\Admin
```

And I would also change the "admin" prefix on all route names, because some of those can easily conflict too:

```
admin.dashboard
admin.login
admin.logout
admin.media-library
admin.password
admin.users
admin.impersonate
```

**Comment by yanhao-li** [comment]
We agree this will be a great improvement, and in order to keep backward compatibility, we are thinking to have a variable defaults to "twill" so that developers could upgrade their system easily.

PR are welcomed!

**Comment by madalinignisca** [comment]
I'd vote for such a change. I usually build either the main admin custom, or with Orchid, and Twill is in for the CMS part, with separate users and logins.

From my perspective, Orchid is the Admin panel, and Twill is the CMS part of a monolith containing data management and a cms. Orchid is using 'platform' in code, while Twill 'admin', and does not really make sense for who's the real admin...

**Comment by ifox** [comment]
Hi @madalinignisca! I wanted to confirm this is something we're going to move forward with for our next breaking release, Twill 3.0.

Your setup sounds very interesting. Super curious to hear more about using Orchid in tandem with Twill.

**Reply by madalinignisca** [reply]
My usage is basically simple.

I use both projects as for small-medium projects. They cover 2 separate concerns:

- Twill is the CMS (Website management, pages, blogs, etc.)
- Orchid is data management of the web app (this is not the website). (Probably some use Nova, I found Orchid to fit better with my development philosophy)

So, usually I build monoliths that cover all things in one. It's very easy for me, and allows me to use data from the app in the CMS as well, for any page that benefits of dynamically changing data.

Frontend for visitors, users is decoupled. For the "website", usually on NuxtJS with TailwindCSS (cms is exposed as an api for frontend), and for the web apps with Quasar.

It's all a perfect combination in the end that helps me deliver quick, stable, and avoid all closed source solutions that might lock my hands and embarres me in front of the clients I serve.

**Reply by madalinignisca** [reply]
Ah, probably I'd love both projects to see they would use their names:
- Twill would use `twill` and not `admin`
- Orchid would use `orchid` and not `platform`.
(_Think why some companies, outside of IT world, abuse making us see their name everywhere)

**Comment by mattrabe** [comment]
I have the same exact thought, and am happy to see that this will be included in a future release.

I also use a different package for other admin features, currently using Voyager. Whether you're using Voyager or Orchid or something else, we probably got here for the same reason: Those packages offer a lot of administration features, but do not provide CMS features. So in my mind, Twill is indeed "the CMS". There will be an "admin" interface for the CMS - but there may be other admin interfaces for other things... Agree that using "twill" instead of "admin" by default would be the best - and ideally let it be configurable.

Thanks for a great package, looking forward to working with it more!
